### DISCUSSION TITLE
Datoviz development progress report

### DISCUSSION THREAD

**Discussion by rossant** [datoviz/datoviz#31] [Announcements]
We'll use this thread to communicate about the current development status.

## History

* Spring 2025: release of Datoviz v0.3, introducing a new Pythonic API built on top of the raw `ctypes` bindings of the C API
* Summer 2024: release of Datoviz v0.2, the first version with prebuilt pip wheels for easy cross-platform installation
* Winter 2022–2023: complete rewrite of the existing Datoviz API on top of the new core engine
* August 2022: active development of the new codebase started in another repository
* September 2021: internal refactoring of the visualization engine
* February 2021: first public experimental preview

## Current development status

* Although Datoviz is still a young library, it is now suitable for most users
* The internal rendering protocol will be upgraded to support more features in Datoviz v0.4, currently planned for early 2026

**Comment by oktay-sahinoglu** [comment]
Has the datoviz development stopped?

**Comment by rossant** [comment]
No it has not, Datoviz is in active development, although in a [different repository](https://github.com/datoviz/datoviz-distributed) for now. An internal refactoring is taking place to improve modularity and maintainability.

In the short term, the current version in this repository remains experimental and targets basic usage.
In the medium term, the user API will remain roughly similar while the core implementation will be much more robust.
In the long term, Datoviz will mainly be a rendering backend for the future VisPy 2.0 visualization library, although it will continue to provide a C API for more advanced users.

**Reply by oktay-sahinoglu** [reply]
Thank you.

**Comment by rossant** [comment]
Datoviz v0.3 has just been released.
